webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala

```diff
@@ -0,0 +1,61 @@
+package com.normation.rudder.services.healthcheck
+
+import java.lang.Runtime.getRuntime
+
+import com.normation.errors.IOResult
+import com.normation.rudder.domain.logger.{HealthcheckLoggerPure => logger}
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Critical
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Warning
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Ok
+import zio.UIO
+import zio.ZIO
+import zio.syntax.ToZio
+
+final case class CheckName(value: String)
+
+trait Check {
+  def name: CheckName
+  def run: IOResult[HealthcheckResult]
+}
+
+sealed trait HealthcheckResult { def msg:String }
+object HealthcheckResult {
+  final case class Ok(val msg: String) extends HealthcheckResult
+  final case class Warning(val msg: String) extends HealthcheckResult
+  final case class Critical(val msg: String) extends HealthcheckResult
+}
+
+class HealthcheckService(checks: List[Check]) {
+
+   def runAll: ZIO[Any, Nothing, List[HealthcheckResult]]  = ZIO.foreach(checks) { c =>
+      for {
+        res <- c.run.catchAll { err =>
+                 HealthcheckResult.Critical(
+                   s"A fatal error was encountered when running check '${c.name}': ${err.fullMsg}"
+                 ).succeed
+               }
+        _   <- logHealthcheck(res)
+      } yield res
+    }
+
+  private[this] def logHealthcheck(check: HealthcheckResult): UIO[Unit] = {
+    check match {
+      case c: Critical => logger.error(s"`Healthcheck has detected a critical issue: ${check.msg}")
+      case w: Warning =>  logger.warn(s"Healthcheck check has detected a warning issue: ${check.msg}")
+      case o: Ok =>       logger.debug(s"Healthcheck check has passed: ${check.msg}")
+    }
+  }
+}
+
+final object CheckCoreNumber extends Check{
+  def name: CheckName = CheckName("CPU Cores available")
+  def run: IOResult[HealthcheckResult] = for {
+    availableCores <- IOResult.effect(getRuntime.availableProcessors)
+  } yield {
+    availableCores match {
+      case 0           => Critical("No Core available")
+      case 1           => Warning(s"Only one cores available")
+      case n if n >= 2 => Ok(s"${n} cores available")
```

Comment:
your test is not total here, it should be: 
```
    availableCores match {
      case n if n < 1 => Critical("No Core available")
      case 1          => Warning(s"Only one cores available")
      case _          => Ok(s"${n} cores available")
```

Reply:
In a second time, we will need to work on messages, because we need to provide user with actionnable corrective actions. 
And perhaps we will want to have to level (one for the "chek tab", on for details).
And perhaps, we will want to externalize this messages, but perhaps not.